test/conftest.py

```diff
@@ -298,6 +301,64 @@ def fixture_standby2_controller(session_tmpdir, mysql_standby2, default_backup_s
         controller.stop()
 
 
+@pytest.fixture(scope="function", name="empty_controller")
+def fixture_empty_controller(
+    session_tmpdir, mysql_empty: MySQLConfig, default_backup_site: BackupSiteInfo
+) -> Iterator[tuple[Controller, MySQLConfig]]:
+    controller = build_controller(
+        Controller,
+        default_backup_site=default_backup_site,
+        mysql_config=mysql_empty,
+        session_tmpdir=session_tmpdir,
+    )
+    try:
+        yield controller, mysql_empty
+    finally:
+        controller.stop()
+
+
+@pytest.fixture(scope="function", name="empty_controller_in_small_disk")
+def fixture_empty_controller_in_small_disk(
+    session_tmpdir, mysql_empty: MySQLConfig, default_backup_site: BackupSiteInfo
+) -> Iterator[tuple[Controller, MySQLConfig]]:
+    sub_dir = Path(session_tmpdir().strpath) / "small_disk"
+    with mount_tmpfs(path=sub_dir, size=8) as tmpfs_dir:
+        controller = build_controller(
+            Controller,
+            default_backup_site=default_backup_site,
+            mysql_config=mysql_empty,
+            session_tmpdir=lambda: LocalPath(tmpfs_dir),
+        )
+        try:
```

Comment:
I don't think try..finally work as intended in fixtures, there has been a discussion about that in a recent Friday demo.